src/frontend/flight/handler.rs

```diff
@@ -74,6 +77,14 @@ impl SeafowlFlightHandler {
             self.context.clone()
         };
 
+        if let Some(memory_store) = memory_store {
+            // If metastore was inlined with the query use it for resolving tables/locations
+            ctx = ctx.with_metastore(Arc::new(Metastore::new_from_memory(
```

Comment:
As discussed, we might need to "merge" the in-memory and the real from-config metastore and fall back to the latter metastore if there's a table we can't find in the former, but we can do this in the separate PR if need be